example/reader_util.c

```diff
@@ -1745,6 +1745,29 @@ int ndpi_is_datalink_supported(int datalink_type) {
   }
 }
 
+static bool ndpi_is_valid_vxlan(const struct pcap_pkthdr *header, const u_char *packet, u_int16_t ip_offset, u_int16_t ip_len){
+  if(header->caplen >= ip_offset + ip_len + sizeof(struct ndpi_vxlanhdr)) {
+    u_int32_t vxlan_dst_port  = ntohs(4789);
+    u_int32_t expected_flags = 0x08; /* only one bit should be set in the first byte */
+    struct ndpi_udphdr *udp = (struct ndpi_udphdr *)&packet[ip_offset+ip_len];
+    u_int offset = ip_offset + ip_len + sizeof(struct ndpi_udphdr);
+    struct ndpi_vxlanhdr *vxlan = (struct ndpi_vxlanhdr *)&packet[offset];
+
+    if((udp->dest == vxlan_dst_port || udp->source == vxlan_dst_port) &&
+      ((vxlan->flags & 0xFF) == expected_flags) &&
+      (((vxlan->flags >> 8) & 0xFF) == 0x0) &&
+      ((vxlan->groupPolicy & 0xFFFF) == 0x0) &&
+      (((vxlan->vni >> 24) & 0xFF) == 0x0)) {
```

Comment:
As you can see from CI results, these checks don't work on big-endian machines....

Reply:
Yes, I saw it today's morning 
I'm going to fix that today at night.